### PR TITLE
planner: init at 0.14.91

### PR DESCRIPTION
**Pull request by amiloradovsky**
###### Description of changes

Planner has been removed 2+ years ago, because it wasn't maintained for a long time, but now it seems to be maintained again and doesn't pull up the ancient dependencies anymore.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/planner/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib, stdenv, fetchurl, meson, ninja, cmake, pkg-config,
+  desktop-file-utils, shared-mime-info, makeWrapper,
+  glib, gtk3, libxml2, libxslt, libgda, gsettings-desktop-schemas }:
+
+stdenv.mkDerivation rec {
+  pname = "planner";
+  version = "0.14.91";
+
+  src = fetchurl {
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:9]: Why not `fetchFromGitLab`?

File: pkgs/applications/office/planner/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    url = "https://gitlab.gnome.org/World/${pname}/-/archive/${version}/${pname}-${version}.tar.bz2";
+    sha256 = "1hd2adm6sg0dqwv32ykw4xv3q7x4hhjckxcx8p55amvvsy3jacy5";
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:11]: Use SRI hash.

File: pkgs/applications/office/planner/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  postPatch = ''patchShebangs \
+                  meson_post_install.sh \
+                  tools/strip_trailing_white_space.sh \
+                  tests/python/task-test.py'';  # grep '^#! */' -R .
+
+  enableParallelBuilding = true;
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:19]: Already the default.

```suggestion
```

File: pkgs/applications/office/planner/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  nativeBuildInputs = [ meson ninja cmake pkg-config
+                        desktop-file-utils shared-mime-info
+                        makeWrapper ];
+  buildInputs = [ glib gtk3 libxml2 libxslt libgda
+                  gsettings-desktop-schemas ];
+
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:25]:
```suggestion
  nativeBuildInputs = [ 
    desktop-file-utils
    cmake
    makeWrapper
    meson
    ninja
    pkg-config
    shared-mime-info
  ];

  buildInputs = [ 
    libgda
    libxml2 
    libxslt
    glib
    gsettings-desktop-schemas
    gtk3 
  ];

```

Comment (AndersonTorres) [reply pkgs/applications/office/planner/default.nix:25]: Why both cmake and meson?

File: pkgs/applications/office/planner/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/${pname}-${version}"
+  '';
+
+  meta = with lib; {
+    description = "Project management tool for the GNOME desktop";
+    homepage    = "https://wiki.gnome.org/Apps/Planner";
+    license     = with licenses; [ gpl2Plus ];
+    platforms   = platforms.all;
+    maintainers = with maintainers; [ amiloradovsky ];
+  };
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:37]:
```suggestion
  meta = {
    description = "Project management tool for the GNOME desktop";
    homepage = "https://wiki.gnome.org/Apps/Planner";
    license = lib.licenses.gpl2Plus;
    maintainers = with lib.maintainers; [ amiloradovsky ];
    platforms = lib.platforms.unix;
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/office/planner/default.nix (hunk 4 above)
Comment (amiloradovsky) [reply pkgs/applications/office/planner/default.nix:24]: I don't remember: but yes it's not needed.

File: pkgs/applications/office/planner/default.nix (hunk 2 above)
Comment (amiloradovsky) [reply pkgs/applications/office/planner/default.nix:11]: SRI hash? How do I get it?

File: pkgs/applications/office/planner/default.nix (hunk 1 above)
Comment (amiloradovsky) [reply pkgs/applications/office/planner/default.nix]: done

File: pkgs/applications/office/planner/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/applications/office/planner/default.nix]: `nix hash to-sri sha256:<old format here>`

**Comment by amiloradovsky** [comment]
(Not sure what else GitHub wants from me by saying "Changes requested" in red: I've made all the edits.)

File: pkgs/applications/office/planner/default.nix [hunk 6]
```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, fetchFromGitLab, meson, ninja, pkg-config,
+  desktop-file-utils, shared-mime-info, makeWrapper,
+  glib, gtk3, libxml2, libxslt, libgda, gsettings-desktop-schemas }:
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:3]: One per line:

```suggestion
{ lib
, stdenv
, fetchFromGitLab

, desktop-file-utils
, gsettings-desktop-schemas
, glib
, gtk3
, libgda
, libxml2
, libxslt
, makeWrapper
, meson
, ninja
, pkg-config
, shared-mime-info
}:
```

File: pkgs/applications/office/planner/default.nix [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "planner";
+  version = "0.14.91";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = pname;
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:12]:
```suggestion
    repo = "planner";
```

File: pkgs/applications/office/planner/default.nix [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    rev = version;
+    hash = "sha256:1xmariry1dwpg5sv8938wskzxvqknifsl1wfsi4qjan2zxk2s5rg";
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:14]: Use SRI:
```suggestion
    hash = "sha256-LxctZv/CKolJ1I4Hql20E+/+p+ZoJLR1eZe34HPMqvY=";
```

File: pkgs/applications/office/planner/default.nix [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  postPatch = ''patchShebangs \
+                  meson_post_install.sh \
+                  tools/strip_trailing_white_space.sh \
+                  tests/python/task-test.py'';  # grep '^#! */' -R .
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:20]:
```suggestion
  postPatch = ''
    patchShebangs\
      meson_post_install.sh \
      tools/strip_trailing_white_space.sh \
      tests/python/task-test.py
  '';
```

File: pkgs/applications/office/planner/default.nix [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    libxml2
+    libxslt
+    glib
+    gsettings-desktop-schemas
+    gtk3
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/planner \
+      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/${pname}-${version}"
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:42]:
```suggestion
      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/planner-${version}"
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Squash the commits into one.

**Review by AndersonTorres — APPROVED** [review]

**Comment by amiloradovsky** [comment]
after two weens the aarch64-darwin test is neither completed nor failed, maybe I should just exclude it from the supported platforms and that would be it, and also do this for all the potential future packages

**Comment by amiloradovsky** [comment]
and it's still waiting for aarch64-darwin, well then

File: pkgs/applications/office/planner/default.nix [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:$out/share/gsettings-schemas/planner-${version}"
+  '';
+
+  meta = {
+    description = "Project management tool for the GNOME desktop";
+    homepage = "https://wiki.gnome.org/Apps/Planner";
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ amiloradovsky ];
+    platforms = lib.platforms.linux;
+    # could be unix^ but the CI tests neither fail nor complete for two weeks
```
Comment (AndersonTorres) [inline pkgs/applications/office/planner/default.nix:66]: Let's be **bold** here:

The Darwin machines are always slow when compiling. Please put `unix` here and let's merge afterwards.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/office/planner/default.nix (hunk 11 above)
Comment (amiloradovsky) [reply pkgs/applications/office/planner/default.nix]: done

**Pull request merged**